web/webkit/src/main/scala/net/liftweb/http/LiftSession.scala

```diff
@@ -2325,32 +2326,68 @@ class LiftSession(private[http] val _contextPath: String, val underlyingId: Stri
    * the CometActor is instantiated.  If the CometActor already exists
    * in the session, the message will be sent immediately.  If the CometActor
    * is not yet instantiated, the message will be sent to the CometActor
-   * as part of setup (@see setupComet) if it is created as part
+   * as part of setup (@see queueCometMessage) if it is created as part
    * of the current HTTP request/response cycle.
    *
    * @param theType the type of the CometActor
+   * @param msg the message to send to the CometActor
+   */
+  def sendCometActorMessage(theType: String, msg: Any) {
+    testStatefulFeature {
+      findComet(theType).foreach(_ ! msg)
+      queueCometMessage(theType, msg)
+    }
+  }
+
+  /**
+   * This method will send a message to a CometActor, whether or not
+   * the CometActor is instantiated.  If the CometActor already exists
+   * in the session, the message will be sent immediately.  If the CometActor
+   * is not yet instantiated, the message will be sent to the CometActor
+   * as part of setup (@see queueCometMessage) regardless of if it is created as part
+   * of the current HTTP request/response cycle. We plan to restrict this to
+   * the current request/response cycle in the future (as that is the intended beavior).
+   *
+   * @param theType the type of the CometActor
    * @param name the optional name of the CometActor
    * @param msg the message to send to the CometActor
    */
   def sendCometActorMessage(theType: String, name: Box[String], msg: Any) {
     testStatefulFeature {
       findComet(theType, name) match {
         case Full(a) => a ! msg
-        case _ => setupComet(theType, name, msg)
+        case _ => queueCometMessage(theType, name, msg)
       }
     }
   }
 
   /**
    * Allows you to send messages to a CometActor that may or may not be set up yet
    */
-  def setupComet(cometType: String, cometName: Box[String], msg: Any) {
+  def queueCometMessage(cometType: String, msg: Any) {
```

Comment:
> if you call queueCometMessage and the comet already existed, will it receive the message?

Ah, probably not. Right you are, sir.

> Also, it feels like we should settle on either sendCometMessage/queueCometMessage or sendCometActorMessage/queueCometActorMessage, maybe?

I like the former since "Comet" implies "Actor" for Lift's purposes.